samples/SocialSample/Startup.cs

```diff
@@ -58,6 +61,17 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
             {
                 options.ClientId = "560027070069-37ldt4kfuohhu3m495hk2j4pjp92d382.apps.googleusercontent.com";
                 options.ClientSecret = "n2Q-GEw9RQjzcRbU3qhfTj8f";
+                options.Events = new OAuthEvents()
+                {
+                    OnRemoteError = ctx =>
```

Comment:
There's a difference between exceptions and protocol errors. Unexpected exceptions will already bubble up, protocol errors need more direct handling (though you could choose to throw in OnRemoteError).